internal/setup/setup.go

```diff
@@ -21,10 +21,9 @@ import (
 // Release is a collection of package slices targeting a particular
 // distribution version.
 type Release struct {
-	Path           string
-	Packages       map[string]*Package
-	Archives       map[string]*Archive
-	DefaultArchive string
```

Comment:
Instead of deprecating, dropping, and ignoring a default archive setting, shouldn't we be supporting it and making it incompatible with priorities?  That is, if we have a default archive, we simply translate that into priorities (e.g. priority=1 for the default, -1 for everyone else). Result should be the same, right?

Note that if we don't do something like that, there's a window of breakage where people update their chisel releases, and their cached revisions do not have the new language.

Reply:
> Instead of deprecating, dropping, and ignoring a default archive setting, shouldn't we be supporting it and making it incompatible with priorities?

When we discussed this in the past, the idea was to keep both "default" and "priority" in chisel releases so both the old versions of Chisel and the new ones will work, old ones will ignore "priority" and new ones will ignore "default". I think that addresses your concern below:

> Note that if we don't do something like that, there's a window of breakage where people update their chisel releases, and their cached revisions do not have the new language.

Reply:
> When we discussed this in the past, the idea was to keep both "default" and "priority" in chisel releases so both the old versions of Chisel and the new ones will work, old ones will ignore "priority" and new ones will ignore "default". I think that addresses your concern below:

Okay, that sounds reasonable. With that said, we can probably do better than ignoring the default field completely, if it in fact needs to remain present otherwise things will break for people.  We can, for example, enforce its uniqueness, and that it is set on the highest priority archive, so that the behavior remains proper for the older releases.

Reply:
It's also worth noting that we cannot verbally flag this as deprecated for existing releases. We've just tagged 1.0, so this needs to remain current while we don't bump the format.  We can bump it for new releases, though, as compatibility breakage is expected across major releases.  Might be worth waiting for a few more things before bothering with it, though.

Reply:
> With that said, we can probably do better than ignoring the default field completely, if it in fact needs to remain present otherwise things will break for people. We can, for example, enforce its uniqueness, and that it is set on the highest priority archive, so that the behavior remains proper for the older releases.

That makes sense and would be helpful to validate the releases. The only problem that I see with trying to match default and priority is that the behavior is not going to be the same anyway. Even if the highest priority is the default archive, the difference is that now we will look in other archives while older Chisel releases will only look at the default. For example, if we add the fips archive, we will add it with the highest priority and it will not be problematic because Chisel will look into the other archives for packages not present there. However, if we also mark it as the default then all the older releases will only look into fips and thus fail to download packages.

Maybe for validation instead of parsing the default field in v1, we could rely on the CI in chisel-releases to check compatibility against several versions: latest commit, v1, v0.7, etc. instead of only the latest commit (what it does now).

Reply:
Discussed offline,

> That is, if we have a default archive, we simply translate that into priorities (e.g. priority=1 for the default, -1 for everyone else)

Is a good idea to prevent breaking people who update to the latest version of Chisel but have not updated their local chisel release (yet).